From 56b741c8793c6058187650aace28231a6188b72d Mon Sep 17 00:00:00 2001
From: Qingsong Chen <changxian.cqs@antgroup.com>
Date: Tue, 28 Jul 2026 20:41:14 +0800
Subject: [PATCH 1/2] Make jtreg use tools from PATH

`jtreg` hardcodes many helper tools as `/bin/*` or `/usr/bin/*`. Those
paths do not exist in the Nix build sandbox, so the build fails before
it can produce the image. Use tool names instead and rely on `PATH`,
which the Nix build environment already populates.
---
 make/Defs.gmk  | 98 +++++++++++++++++---------------------------------
 make/jtreg.gmk |  2 +-
 2 files changed, 33 insertions(+), 67 deletions(-)

diff --git a/make/Defs.gmk b/make/Defs.gmk
index d78d5e7..a4cd27b 100644
--- a/make/Defs.gmk
+++ b/make/Defs.gmk
@@ -134,65 +134,31 @@ REGTEST_TOOL_JAVAC_OPTIONS = \
 
 #----- Unix commands
 
-AWK     = /usr/bin/awk
-CAT 	= /bin/cat
-CHMOD 	= /bin/chmod
-CP 	= /bin/cp
-DIFF 	= /usr/bin/diff
-ECHO	= /bin/echo
-FIND	= /usr/bin/find
-ifeq ($(SYSTEM_UNAME), FreeBSD)
-GREP  := $(shell if [ -r /usr/local/bin/ggrep ]; then \
-    echo /usr/local/bin/ggrep ; \
-  else \
-    echo $(ECHO) "GNU grep is required, but was not found" ; \
-  fi )
-else
-GREP 	:= $(shell if [ -r /bin/grep ]; then echo /bin/grep ; else echo /usr/bin/grep ; fi )
-endif
-LN	= /bin/ln
-LS	= /bin/ls
-MKDIR 	= /bin/mkdir -p
-MV 	= /bin/mv
-PANDOC  := $(shell if [ -r /usr/bin/pandoc ]; then \
-		echo /usr/bin/pandoc ; \
-	elif [ -r /usr/local/bin/pandoc ]; then \
-		echo /usr/local/bin/pandoc ; \
-	elif [ -r /opt/homebrew/bin/pandoc ]; then \
-		echo /opt/homebrew/bin/pandoc ; \
-	else \
-		echo /bin/echo "pandoc not available" ; \
-	fi )
-PERL	= /usr/bin/perl
-PRINTF  = /usr/bin/printf
-RM 	= /bin/rm -rf
-SED 	:= $(shell if [ -r /bin/sed ]; then echo /bin/sed ; else echo /usr/bin/sed ; fi )
-SH 	= /bin/sh
-SORT    = /usr/bin/sort
-TEST 	= /usr/bin/test
-ifeq ($(SYSTEM_UNAME), Darwin)
-TIDY 	:= $(shell if [ -r /usr/local/bin/tidy ]; then \
-		echo /usr/local/bin/tidy ; \
-	elif [ -r /opt/homebrew/bin/tidy ]; then \
-		echo /opt/homebrew/bin/tidy ; \
-	else \
-		echo /usr/bin/tidy ; \
-	fi )
-else
-TIDY	= /usr/bin/tidy
-endif
-TOUCH 	= /usr/bin/touch
-ifeq ($(SYSTEM_UNAME), FreeBSD)
-UNZIP	= /usr/local/bin/unzip
-else
-UNZIP 	= /usr/bin/unzip
-endif
-WC 	= /usr/bin/wc
-ifeq ($(SYSTEM_UNAME), FreeBSD)
-ZIP	= /usr/local/bin/zip
-else
-ZIP 	= /usr/bin/zip
-endif
+AWK     = awk
+CAT 	= cat
+CHMOD 	= chmod
+CP 	= cp
+DIFF 	= diff
+ECHO	= echo
+FIND	= find
+GREP 	:= grep
+LN	= ln
+LS	= ls
+MKDIR 	= mkdir -p
+MV 	= mv
+PANDOC  := pandoc
+PERL	= perl
+PRINTF  = printf
+RM 	= rm -rf
+SED 	:= sed
+SH 	= sh
+SORT    = sort
+TEST 	= test
+TIDY	= tidy
+TOUCH 	= touch
+UNZIP 	= unzip
+WC 	= wc
+ZIP 	= zip
 
 
 #----------------------------------------------------------------------
@@ -228,12 +194,12 @@ BUILD_MILESTONE = dev
 BUILD_NUMBER = b00
 
 # don't eval dates here directly, because that leads to unstable builds
-#BUILD_YEAR:sh = /bin/date +"%Y"
-BUILD_YEAR_CMD = /bin/date '+%Y'
-#BUILD_DOCDATE:sh = /bin/date +"%B %d, %Y"
-BUILD_DOCDATE_CMD = /bin/date  +'%B %d, %Y'
-#BUILD_ZIPDATE:sh = /bin/date '+%d %h %Y'
-BUILD_ZIPDATE_CMD = /bin/date  '+%d %h %Y'
+#BUILD_YEAR:sh = date +"%Y"
+BUILD_YEAR_CMD = date '+%Y'
+#BUILD_DOCDATE:sh = date +"%B %d, %Y"
+BUILD_DOCDATE_CMD = date  +'%B %d, %Y'
+#BUILD_ZIPDATE:sh = date '+%d %h %Y'
+BUILD_ZIPDATE_CMD = date  '+%d %h %Y'
 BUILD_NONFCS_MILESTONE_sh = echo $(BUILD_MILESTONE) | sed -e 's/[fF][cC][sS]//'
 BUILD_NONFCS_MILESTONE = $(BUILD_NONFCS_MILESTONE_sh:sh)
 
@@ -242,7 +208,7 @@ ZIPSFX_VERSION_sh = echo '$(BUILD_VERSION)' | sed -e 's|\([^0-9][^0-9]*\)|_|g'
 ZIPSFX_MILESTONE_sh = echo '$(BUILD_MILESTONE)'
 ZIPSFX_BUILD_sh = echo '$(BUILD_NUMBER)'
 ZIPSFX_NEWBUILD_sh = echo '$(BUILD_NUMBER)' | sed -e 's|[^[0-9]||g' | xargs printf "%d"
-ZIPSFX_DATE_sh = echo "`$(BUILD_ZIPDATE_CMD)`" | /usr/bin/tr -s '[A-Z] ' '[a-z]_'
+ZIPSFX_DATE_sh = echo "`$(BUILD_ZIPDATE_CMD)`" | tr -s '[A-Z] ' '[a-z]_'
 
 VERBOSE_ZIP_SUFFIX = $(shell $(ZIPSFX_VERSION_sh))-$(shell $(ZIPSFX_MILESTONE_sh))-bin-$(shell $(ZIPSFX_BUILD_sh))-$(shell $(ZIPSFX_DATE_sh))
 
diff --git a/make/jtreg.gmk b/make/jtreg.gmk
index 136e091..bf03c60 100644
--- a/make/jtreg.gmk
+++ b/make/jtreg.gmk
@@ -345,7 +345,7 @@ ID_COMMAND := $(PRINTF) "git:%s%s\n" \
 
 $(JTREG_IMAGEDIR)/release:
 	echo "JTREG_VERSION=$(BUILD_VERSION) $(BUILD_NUMBER)" > $@
-	echo "BUILD_DATE=`/bin/date +'%B %d, %Y'`" >> $@
+	echo "BUILD_DATE=`date +'%B %d, %Y'`" >> $@
 	if [ -r $(TOPDIR)/.git ]; then \
 	    echo "SOURCE=$$($(ID_COMMAND))" >> $@ ; \
 	elif [ -n "$(SRCREV)" -a -r $(SRCREV) ]; then \
-- 
2.54.0

